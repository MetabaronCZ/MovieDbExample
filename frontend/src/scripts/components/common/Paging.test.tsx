import React from 'react';

import { act } from '@testing-library/react';
import { describe, expect, it, jest } from '@jest/globals';
import {
  fireEvent,
  getByText,
  getByTitle,
  queryByRole,
  getAllByRole,
} from '@testing-library/dom';

import { Paging } from './Paging';
import { PagingConfig, usePaging } from 'hooks/usePaging';

import { flushPromises } from 'test-utils/core';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

interface Props {
  readonly config: PagingConfig<number>;
}

const PagingWrapper: React.FC<Props> = ({ config }) => {
  const paging = usePaging(config);
  return (
    <TestComponentWrapper>
      <Paging paging={paging} />
    </TestComponentWrapper>
  );
};

describe('components/common/Paging', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(
      <PagingWrapper
        config={{
          totalCount: 10,
          perPage: 3,
          onPage: () => Promise.resolve(),
          onPerPage: () => Promise.resolve(),
        }}
      />,
    );
    expect(container).toMatchSnapshot();

    const pagingInfo = getByText(container, 'Položky 1 až 3 z 10');
    expect(pagingInfo).toBeInTheDocument();

    const pageNumber = getByText(container, '1');
    expect(pageNumber).toBeInTheDocument();

    const buttonToFirst = getByTitle(container, 'Na začátek');
    expect(buttonToFirst).toBeInTheDocument();
    expect(buttonToFirst.tagName.toLowerCase()).toEqual('button');

    const buttonToLast = getByTitle(container, 'Na konec');
    expect(buttonToLast).toBeInTheDocument();
    expect(buttonToLast.tagName.toLowerCase()).toEqual('button');

    const buttonPrev = getByTitle(container, 'Předchozí');
    expect(buttonPrev).toBeInTheDocument();
    expect(buttonPrev.tagName.toLowerCase()).toEqual('button');

    const buttonNext = getByTitle(container, 'Další');
    expect(buttonNext).toBeInTheDocument();
    expect(buttonNext.tagName.toLowerCase()).toEqual('button');

    const pagingLabel = getByText(container, 'Stránkování');
    expect(pagingLabel).toBeInTheDocument();

    const htmlFor = pagingLabel.getAttribute('for') ?? '';
    expect(htmlFor).not.toEqual('');

    const pagingHandle = container.querySelector(`[id="${htmlFor}"]`);
    expect(pagingHandle).toBeInTheDocument();
    expect(pagingHandle).toHaveTextContent('3'); // selected paging
  });

  it('should render first page', async () => {
    const { container } = await waitForComponent(
      <PagingWrapper
        config={{
          totalCount: 10,
          perPage: 3,
          page: 0,
          onPage: () => Promise.resolve(),
          onPerPage: () => Promise.resolve(),
        }}
      />,
    );

    const pagingInfo = getByText(container, 'Položky 1 až 3 z 10');
    expect(pagingInfo).toBeInTheDocument();

    const pageNumber = getByText(container, '1');
    expect(pageNumber).toBeInTheDocument();

    const buttonToFirst = getByTitle(container, 'Na začátek');
    expect(buttonToFirst).toBeInTheDocument();
    expect(buttonToFirst).toHaveAttribute('disabled', '');

    const buttonToLast = getByTitle(container, 'Na konec');
    expect(buttonToLast).toBeInTheDocument();
    expect(buttonToLast).not.toHaveAttribute('disabled', '');

    const buttonPrev = getByTitle(container, 'Předchozí');
    expect(buttonPrev).toBeInTheDocument();
    expect(buttonPrev).toHaveAttribute('disabled', '');

    const buttonNext = getByTitle(container, 'Další');
    expect(buttonNext).toBeInTheDocument();
    expect(buttonNext).not.toHaveAttribute('disabled', '');
  });

  it('should render inner page', async () => {
    const { container } = await waitForComponent(
      <PagingWrapper
        config={{
          totalCount: 10,
          perPage: 3,
          page: 1,
          onPage: () => Promise.resolve(),
          onPerPage: () => Promise.resolve(),
        }}
      />,
    );

    const pagingInfo = getByText(container, 'Položky 4 až 6 z 10');
    expect(pagingInfo).toBeInTheDocument();

    const pageNumber = getByText(container, '2');
    expect(pageNumber).toBeInTheDocument();

    const buttonToFirst = getByTitle(container, 'Na začátek');
    expect(buttonToFirst).toBeInTheDocument();
    expect(buttonToFirst).not.toHaveAttribute('disabled', '');

    const buttonToLast = getByTitle(container, 'Na konec');
    expect(buttonToLast).toBeInTheDocument();
    expect(buttonToLast).not.toHaveAttribute('disabled', '');

    const buttonPrev = getByTitle(container, 'Předchozí');
    expect(buttonPrev).toBeInTheDocument();
    expect(buttonPrev).not.toHaveAttribute('disabled', '');

    const buttonNext = getByTitle(container, 'Další');
    expect(buttonNext).toBeInTheDocument();
    expect(buttonNext).not.toHaveAttribute('disabled', '');
  });

  it('should render last page', async () => {
    const { container } = await waitForComponent(
      <PagingWrapper
        config={{
          totalCount: 10,
          perPage: 3,
          page: 3,
          onPage: () => Promise.resolve(),
          onPerPage: () => Promise.resolve(),
        }}
      />,
    );

    const pagingInfo = getByText(container, 'Položky 10 až 10 z 10');
    expect(pagingInfo).toBeInTheDocument();

    const pageNumber = getByText(container, '4');
    expect(pageNumber).toBeInTheDocument();

    const buttonToFirst = getByTitle(container, 'Na začátek');
    expect(buttonToFirst).toBeInTheDocument();
    expect(buttonToFirst).not.toHaveAttribute('disabled', '');

    const buttonToLast = getByTitle(container, 'Na konec');
    expect(buttonToLast).toBeInTheDocument();
    expect(buttonToLast).toHaveAttribute('disabled', '');

    const buttonPrev = getByTitle(container, 'Předchozí');
    expect(buttonPrev).toBeInTheDocument();
    expect(buttonPrev).not.toHaveAttribute('disabled', '');

    const buttonNext = getByTitle(container, 'Další');
    expect(buttonNext).toBeInTheDocument();
    expect(buttonNext).toHaveAttribute('disabled', '');
  });

  it('should be navigable', async () => {
    const { container } = await waitForComponent(
      <PagingWrapper
        config={{
          totalCount: 10,
          perPage: 3,
          onPage: () => Promise.resolve(),
          onPerPage: () => Promise.resolve(),
        }}
      />,
    );

    // show first page by default
    let pagingInfo = getByText(container, 'Položky 1 až 3 z 10');
    expect(pagingInfo).toBeInTheDocument();

    let pageNumber = getByText(container, '1');
    expect(pageNumber).toBeInTheDocument();

    const buttonToFirst = getByTitle(container, 'Na začátek');
    expect(buttonToFirst).toBeInTheDocument();
    expect(buttonToFirst).toHaveAttribute('disabled', '');

    const buttonToLast = getByTitle(container, 'Na konec');
    expect(buttonToLast).toBeInTheDocument();
    expect(buttonToLast).not.toHaveAttribute('disabled', '');

    const buttonPrev = getByTitle(container, 'Předchozí');
    expect(buttonPrev).toBeInTheDocument();
    expect(buttonPrev).toHaveAttribute('disabled', '');

    const buttonNext = getByTitle(container, 'Další');
    expect(buttonNext).toBeInTheDocument();
    expect(buttonNext).not.toHaveAttribute('disabled', '');

    // go to next page
    await act(async () => {
      fireEvent.click(buttonNext);
      await flushPromises();
    });

    pagingInfo = getByText(container, 'Položky 4 až 6 z 10');
    expect(pagingInfo).toBeInTheDocument();

    pageNumber = getByText(container, '2');
    expect(pageNumber).toBeInTheDocument();

    expect(buttonToFirst).not.toHaveAttribute('disabled', '');
    expect(buttonToLast).not.toHaveAttribute('disabled', '');
    expect(buttonPrev).not.toHaveAttribute('disabled', '');
    expect(buttonNext).not.toHaveAttribute('disabled', '');

    // go to last page
    await act(async () => {
      fireEvent.click(buttonToLast);
      await flushPromises();
    });

    pagingInfo = getByText(container, 'Položky 10 až 10 z 10');
    expect(pagingInfo).toBeInTheDocument();

    pageNumber = getByText(container, '4');
    expect(pageNumber).toBeInTheDocument();

    expect(buttonToFirst).not.toHaveAttribute('disabled', '');
    expect(buttonToLast).toHaveAttribute('disabled', '');
    expect(buttonPrev).not.toHaveAttribute('disabled', '');
    expect(buttonNext).toHaveAttribute('disabled', '');

    // go to first page
    await act(async () => {
      fireEvent.click(buttonToFirst);
      await flushPromises();
    });

    pagingInfo = getByText(container, 'Položky 1 až 3 z 10');
    expect(pagingInfo).toBeInTheDocument();

    pageNumber = getByText(container, '1');
    expect(pageNumber).toBeInTheDocument();

    expect(buttonToFirst).toHaveAttribute('disabled', '');
    expect(buttonToLast).not.toHaveAttribute('disabled', '');
    expect(buttonPrev).toHaveAttribute('disabled', '');
    expect(buttonNext).not.toHaveAttribute('disabled', '');
  });

  it('should contain perPage select', async () => {
    const onPerPage = jest.fn(() => Promise.resolve());

    const { container } = await waitForComponent(
      <PagingWrapper
        config={{
          totalCount: 10,
          perPage: 3,
          perPages: [3, 6, 9],
          onPage: () => Promise.resolve(),
          onPerPage,
        }}
      />,
    );
    expect(container).toMatchSnapshot();
    expect(onPerPage).not.toBeCalled();

    const pagingLabel = getByText(container, 'Stránkování');
    const htmlFor = pagingLabel.getAttribute('for') ?? '';

    const pagingHandle = container.querySelector(`[id="${htmlFor}"]`);
    expect(pagingHandle).toBeInTheDocument();
    expect(pagingHandle).toHaveTextContent('3'); // selected paging

    const selectContainer = pagingHandle?.parentNode;

    if (!pagingHandle || !selectContainer) {
      throw new Error('Could not find perPage select handle!');
    }
    let pagingList = queryByRole(selectContainer as HTMLElement, 'list');
    expect(pagingList).not.toBeInTheDocument();

    // open perPage select
    act(() => {
      fireEvent.click(pagingHandle);
    });

    pagingList = queryByRole(selectContainer as HTMLElement, 'list');
    expect(pagingList).toBeInTheDocument();

    if (!pagingList) {
      throw new Error('Could not find perPage select items container!');
    }
    const items = getAllByRole(pagingList, 'button');
    expect(items.length).toEqual(3);

    expect(items[0]).toHaveTextContent('3');
    expect(items[1]).toHaveTextContent('6');
    expect(items[2]).toHaveTextContent('9');

    // select another perPage
    await act(async () => {
      fireEvent.click(items[2]);
      await flushPromises();
    });
    expect(onPerPage).toBeCalledTimes(1);
    expect(onPerPage).toBeCalledWith(9);
  });
});
