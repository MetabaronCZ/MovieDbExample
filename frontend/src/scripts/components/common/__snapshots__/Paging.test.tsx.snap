// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Paging should contain perPage select 1`] = `
.c3 {
  display: block;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
}

.c3:disabled {
  cursor: not-allowed;
}

.c4 {
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  width: 32px;
  height: 32px;
}

.c4:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c4:focus {
  outline: 1px dashed #002d56;
}

.c11 {
  flex: 1;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c10 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  gap: 12px;
  line-height: 1;
  font-weight: 700;
  height: 40px;
  padding: 0 16px;
  padding-left: 12px;
  padding-right: 12px;
  border: 2px solid #002d56;
  background: #fff;
  width: 100%;
  height: 40px;
  gap: 8px;
  padding-left: 12px;
  padding-right: 12px;
}

.c10:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c10:focus {
  outline: 1px dashed #002d56;
}

.c10:disabled {
  color: #98c3d9;
  border-color: #98c3d9;
  background-color: #fff;
}

.c12 {
  font-size: 14px;
  font-weight: 400;
  text-align: left;
}

.c9 {
  position: relative;
  display: inline-block;
  width: 100%;
}

.c8 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: flex;
  flex-direction: row;
  gap: 4px;
}

.c6 {
  display: flex;
  flex-direction: column;
  gap: 4px;
  border: none;
}

.c7 {
  display: flex;
  flex-direction: row;
  gap: 16px;
  align-items: center;
  justify-content: start;
}

.c0 {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  gap: 8px;
  padding-top: 8px;
  border-top: 1px solid #dae0e3;
}

.c2 {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  gap: 8px;
}

.c1 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 12px;
  line-height: 24px;
  color: #98c3d9;
}

.c5 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  min-width: 24px;
  font-weight: bold;
  text-align: center;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Položky 1 až 3 z 10
    </div>
    <div
      class="c2"
    >
      <button
        class="c3 c4"
        disabled=""
        title="Na začátek"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angles-left fa-sm "
          color="#98c3d9"
          data-icon="angles-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160zm352-160l-160 160c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L301.3 256 438.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="c3 c4"
        disabled=""
        title="Předchozí"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angle-left fa-sm "
          color="#98c3d9"
          data-icon="angle-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
            fill="currentColor"
          />
        </svg>
      </button>
      <div
        class="c5"
      >
        1
      </div>
      <button
        class="c3 c4"
        title="Další"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angle-right fa-sm "
          color="#002d56"
          data-icon="angle-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M278.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-160 160c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L210.7 256 73.4 118.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l160 160z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="c3 c4"
        title="Na konec"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angles-right fa-sm "
          color="#002d56"
          data-icon="angles-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M470.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L402.7 256 265.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160zm-352 160l160-160c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L210.7 256 73.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <fieldset
      class="c6"
    >
      <div
        class="c7"
      >
        <label
          class="c8"
          for="select-16"
        >
          Stránkování
        </label>
        <div
          class="c9"
        >
          <button
            class="c3 c10"
            id="select-16"
            type="button"
          >
            <span
              class="c11 c12"
            >
              3
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-down fa-lg "
              data-icon="angle-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;

exports[`components/common/Paging should render properly 1`] = `
.c3 {
  display: block;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
}

.c3:disabled {
  cursor: not-allowed;
}

.c4 {
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  width: 32px;
  height: 32px;
}

.c4:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c4:focus {
  outline: 1px dashed #002d56;
}

.c11 {
  flex: 1;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c10 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  gap: 12px;
  line-height: 1;
  font-weight: 700;
  height: 40px;
  padding: 0 16px;
  padding-left: 12px;
  padding-right: 12px;
  border: 2px solid #002d56;
  background: #fff;
  width: 100%;
  height: 40px;
  gap: 8px;
  padding-left: 12px;
  padding-right: 12px;
}

.c10:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c10:focus {
  outline: 1px dashed #002d56;
}

.c10:disabled {
  color: #98c3d9;
  border-color: #98c3d9;
  background-color: #fff;
}

.c12 {
  font-size: 14px;
  font-weight: 400;
  text-align: left;
}

.c9 {
  position: relative;
  display: inline-block;
  width: 100%;
}

.c8 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: flex;
  flex-direction: row;
  gap: 4px;
}

.c6 {
  display: flex;
  flex-direction: column;
  gap: 4px;
  border: none;
}

.c7 {
  display: flex;
  flex-direction: row;
  gap: 16px;
  align-items: center;
  justify-content: start;
}

.c0 {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  gap: 8px;
  padding-top: 8px;
  border-top: 1px solid #dae0e3;
}

.c2 {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  gap: 8px;
}

.c1 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 12px;
  line-height: 24px;
  color: #98c3d9;
}

.c5 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  min-width: 24px;
  font-weight: bold;
  text-align: center;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Položky 1 až 3 z 10
    </div>
    <div
      class="c2"
    >
      <button
        class="c3 c4"
        disabled=""
        title="Na začátek"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angles-left fa-sm "
          color="#98c3d9"
          data-icon="angles-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160zm352-160l-160 160c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L301.3 256 438.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="c3 c4"
        disabled=""
        title="Předchozí"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angle-left fa-sm "
          color="#98c3d9"
          data-icon="angle-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
            fill="currentColor"
          />
        </svg>
      </button>
      <div
        class="c5"
      >
        1
      </div>
      <button
        class="c3 c4"
        title="Další"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angle-right fa-sm "
          color="#002d56"
          data-icon="angle-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M278.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-160 160c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L210.7 256 73.4 118.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l160 160z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="c3 c4"
        title="Na konec"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angles-right fa-sm "
          color="#002d56"
          data-icon="angles-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M470.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L402.7 256 265.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160zm-352 160l160-160c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L210.7 256 73.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <fieldset
      class="c6"
    >
      <div
        class="c7"
      >
        <label
          class="c8"
          for="select-1"
        >
          Stránkování
        </label>
        <div
          class="c9"
        >
          <button
            class="c3 c10"
            id="select-1"
            type="button"
          >
            <span
              class="c11 c12"
            >
              3
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-down fa-lg "
              data-icon="angle-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;
